activejob/test/cases/test_helper_test.rb

```diff
@@ -927,7 +927,24 @@ def test_perform_enqueued_jobs_dont_perform_retries
     end
 
     assert_equal(1, performed_jobs.size)
-    assert_equal(2, enqueued_jobs.size)
+    assert_equal(1, enqueued_jobs.size)
```

Comment:
This assertion from  #38686 changes, since the retry is in the enqueued jobs, but no longer the first perform.